firmware/src/hal/sgx/test/secret_store/test_secret_store.c

```diff
@@ -0,0 +1,385 @@
+#include <assert.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "assert_utils.h"
+#include "secret_store.h"
+#include "mock_seal.h"
+#include "mock_ocall.h"
+
+// Error code for the sest API as defined in secret_store.c
+#define SEST_ERROR (0)
+// The maximum value that can be returned by sest_read
+#define MAX_SEST_READ_SIZE (255)
+
+void setup() {
+    mock_seal_init();
+    mock_ocall_init();
+    assert(sest_init());
+}
+
+void test_secret_exists_after_write() {
+    setup();
+    printf("Test secret exists after write...\n");
+
+    char* key = "key";
+    uint8_t secret[] = "secret";
+    // Ensure the secret doesn't exist before the write
+    assert(!sest_exists(key));
+    // Write the secret and ensure it now exists
+    assert(sest_write("key", secret, sizeof(secret)));
+    assert_oe_seal_called_with(
+        NULL,
+        (const oe_seal_setting_t[]){OE_SEAL_SET_POLICY(1)},
+        1,
+        secret,
+        sizeof(secret),
+        NULL,
+        0);
```

Comment:
Shouldn't we probably also assert that the correct sealed secret has been written to the key-value store?

Reply:
Sounds good, will add this check